src/arch/Lights/LightsDriver_LinuxPacDrive.cpp

```diff
@@ -163,120 +173,153 @@ void LightsDriver_LinuxPacDrive::Set( const LightsState *ls )
 
 void LightsDriver_LinuxPacDrive::FindDevice()
 {
-	if ( usb_find_busses() < 0 )
-	{
-		LOG->Warn( "libusb: usb_find_busses: %s", usb_strerror() );
-		return;
-	}
-
-	if ( usb_find_devices() < 0 )
-	{
-		LOG->Warn( "libusb: usb_find_devices: %s", usb_strerror() );
-		return;
-	}
-
-	for ( usb_bus *bus = usb_get_busses(); bus; bus = bus->next )
-		for ( struct usb_device *dev = bus->devices; dev; dev = dev->next )
-			if ( PACDRIVE_VENDOR_ID == dev->descriptor.idVendor &&
-				 PACDRIVE_PRODUCT_ID <= dev->descriptor.idProduct &&
-				 PACDRIVE_PRODUCT_ID + 8 > dev->descriptor.idProduct ) {
-				Device = dev;
-				LOG->Info( "PacDrive device was found vid: 0x%04x pid: 0x%04x", dev->descriptor.idVendor, dev->descriptor.idProduct );
-				return;
-			}
-
-	LOG->Warn( "PacDrive was not found!" );
-	Device = NULL;
+    libusb_device **devs;
+    libusb_device *dev;
+    int i = 0;
+
+    libusb_context* context = USBContext::getInstance().getContext();
+    if (!context)
+    {
+        LOG->Warn("libusb: Failed to initialize context");
+        return;
+    }
+
+    ssize_t cnt = libusb_get_device_list(context, &devs);
+    if (cnt < 0)
+    {
+        LOG->Warn("libusb: No devices found");
+        return;
+    }
+
+    while ((dev = devs[i++]) != NULL)
+    {
+        struct libusb_device_descriptor desc;
+        int r = libusb_get_device_descriptor(dev, &desc);
+        if (r < 0)
```

Comment:
if we unable to get a device descriptor then .. is it so critical that we immediately return from a func instead of trying next device? during return do we need any cleanup?

Reply:
It seems like this is how libusb intends for the developer to handle such a case,  because it might return 0 even if some kind of a device is there.

If i'm wrong about this, then I'm glad to be corrected.

This is my reference:
https://github.com/libusb/libusb/issues/688